cshared.go

```diff
@@ -170,6 +171,8 @@ func FLBPluginInit(ptr unsafe.Pointer) int {
 		}
 		err = theOutput.Init(ctx, fbit)
 	} else {
+		// intended to longer liveness for custom plugin
+		runCancel = cancel
```

Comment:
If this sets `runCancel`, should it also set `runCtx = ctx`?

More generally, the use and lifetimes of `runCtx` and `runCancel` are a little confusing to me. They're global but can be set from a few places with different values. And they're set to nil sometimes. It's worth adding docs to the code on the behavior and invariants of the globals.

Reply:
I think setting `runCtx` should be enough in the scope of this PR.

Both `runCtx` and `runCancel` are defined in `plugin.go`, where we should document it. I am not much of a fan of how it is designed right now, which will need to be overhauled to support multiple instances of plugins.

Reply:
`runCtx` could be used for flush and input callback. In this case, it's intended to use for custom plugins. So, it needn't set up for `runCtx`. Just handling `runCancel` is enough, I think.